hoomd/hpmc/tune/nec_chain_time.py

```diff
@@ -0,0 +1,286 @@
+        # _ChainTimeTuneDefinition returns y == target for that case, we need
+        # two rounds of tuning to be sure that we have converged. Since, in
+        # general, solvers do not do much if any work on already tuned tunables,
+        # this is not a performance problem.
+        self._tuned = 0
+        self._is_attached = False
+
+        self._chain_time_def = _ChainTimeTuneDefinition(
+            target, (self._min_chain_time, max_chain_time))
+
+        param_dict = ParameterDict(
+            target=OnlyTypes(float,
+                             postprocess=self._process_chain_time_target),
+            solver=SolverStep,
+            max_chain_time=OnlyTypes(
+                float,
+                allow_none=True,
+                postprocess=self._process_chain_time_range),
+            min_chain_time=OnlyTypes(
+                float, postprocess=self._process_chain_time_range))
+
+        self._param_dict.update(param_dict)
+        self.target = target
+        self.solver = solver
+        self.max_chain_time = max_chain_time
+        self.min_chain_time = self._min_chain_time
+
+    def _process_chain_time_range(self, target):
+        # check range
+        if 0 <= target:
+            pass
+        else:
+            raise ValueError(
+                "Chain time {} should be between 0 and 1000.".format(target))
+        self._tuned = 0
+        return target
+
+    def _process_chain_time_target(self, target):
+        # check range
+        if 0 <= target <= 1000:
+            pass
+        else:
+            raise ValueError(
+                "Value {} should be between 0 and 1000.".format(target))
+
+        self._chain_time_def.target = target
+        self._tuned = 0
+        return target
+
+    def attach(self, simulation):
+        if not isinstance(simulation.operations.integrator, HPMCNECIntegrator):
+            raise RuntimeError(
+                "ChainTimeTuner can only be used in HPMC-NEC simulations.")
+        self._chain_time_def.integrator = simulation.operations.integrator
+        self._is_attached = True
+
+    @property
+    def _attached(self):
+        """bool: Whether or not the tuner is attached to a simulation."""
+        return self._is_attached
+
+    @property
+    def tuned(self):
+        """bool: Whether or not the move sizes are considered tuned.
+
+        A `ChainTime` object is considered tuned if it the solver tolerance has
+        been met by all tunables for 2 iterations.
+        """
+        return self._tuned >= 2
+
+    def detach(self):
+        self._chain_time_def.integrator = None
+        self._is_attached = False
+
+    def act(self, timestep=None):
+        """Tune chain time.
+
+        Args:
+            timestep (`int`, optional): Current simulation timestep. Is
+                currently ignored.
+        """
+        if self._is_attached:
+            # update maximum chain time
+            if self._update_chain_time:
```

Comment:
This is always `False`. I think some logic is missing that sets the `maximum_chain_size` which could go in the `_process_chain_time` method.

Reply:
If you know how to fix, please do so.